### PR TITLE
Add tests and enable travis

### PR DESCRIPTION
**Pull request by msullivan**
The tests (and a bunch of the config) is pulled in from mypy

### CODE REVIEW

**Comment by msullivan** [comment]
The travis was tested on my fork at https://travis-ci.org/msullivan/mypy_extensions, but needs an admin to enable it on the python org.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    env:
+    - TOXENV=lint
+
+install:
+- pip install -U pip setuptools
+- pip install -U tox
+- tox --notest
+
+script:
+- tox
```
Comment (ilevkivskyi) [inline .travis.yml]: Since this is a "runtime" package, I would rather use the Travis settings from `python/typing`.

Comment (msullivan) [reply .travis.yml]: Which things from typing do you prefer? Is it just running on a bunch more versions, or do you want to not use tox?

Comment (ilevkivskyi) [reply .travis.yml]: Actually both (but first is probably more important).

Comment (msullivan) [reply .travis.yml]: What's the reason to avoid tox? (I don't have any real opinion one way or another except that the tox version works already :P)

Comment (ilevkivskyi) [reply .travis.yml]: I just don't want to learn something the need for what I don't understand. But since it already works, then OK.

**Pull request merged**